src/Resistor/Resistor.ts

```diff
@@ -0,0 +1,48 @@
+import ResistorColorEntry from './ResistorColorEntry';
+
+class Resistor {
+  // tslint:disable max-line-length
+  //                                                                         Name      Color      Multiplier Value Tolerance
+  public static readonly BLACK: ResistorColorEntry =  new ResistorColorEntry('Black',  '#000000', 1,            0, undefined);
+  public static readonly BROWN: ResistorColorEntry =  new ResistorColorEntry('Brown',  '#000000', 10,           1, '1%');
+  public static readonly RED: ResistorColorEntry =    new ResistorColorEntry('Red',    '#FF0000', 100,          2, '2%');
+  public static readonly ORANGE: ResistorColorEntry = new ResistorColorEntry('Orange', '#EE7420', 1000,         3, undefined);
+  public static readonly YELLOW: ResistorColorEntry = new ResistorColorEntry('Yellow', '#FFFF00', 10000,        4, undefined);
+  public static readonly GREEN: ResistorColorEntry =  new ResistorColorEntry('Green',  '#008000', 100000,       5, '0.5%');
+  public static readonly BLUE: ResistorColorEntry =   new ResistorColorEntry('Blue',   '#0000FF', 1000000,      6, '0.25%');
+  public static readonly VIOLET: ResistorColorEntry = new ResistorColorEntry('Violet', '#800080', 10000000,     7, '0.1%');
+  public static readonly GREY: ResistorColorEntry =   new ResistorColorEntry('Grey',   '#808080', 100000000,    8, undefined);
+  public static readonly WHITE: ResistorColorEntry =  new ResistorColorEntry('White',  '#FFFFFF', 1000000000,   9, undefined);
+  public static readonly GOLD: ResistorColorEntry =   new ResistorColorEntry('Gold',   '#AB8D3F', 0.1,  undefined, '5%');
+  public static readonly SILVER: ResistorColorEntry = new ResistorColorEntry('Silver', '#C0C0C0', 0.01, undefined, '10%');
+  // tslint:enable max-line-length
+
+  public static readonly colorTable: ResistorColorEntry[] =
+    [ Resistor.BLACK, Resistor.BROWN, Resistor.RED, Resistor.ORANGE, Resistor.YELLOW, Resistor.GREEN,
+      Resistor.BLUE, Resistor.VIOLET, Resistor.GREY, Resistor.WHITE, Resistor.GOLD, Resistor.SILVER ];
+
+  /** getValue assumes there is no tolerance band, as tolerance is not part of the value calculation */
+  public static getValue(colors: ResistorColorEntry[]) {
+    if (colors.length < 3 || colors.length > 4) {
+      throw new Error('Invalid resistor size');
+    }
+
+    // Should we throw an error if the first color is black?  In resistors, it's technically not allowed, but in puzzle
+    // events who knows what they'll do.  The math works out fine (it's just a leading zero), so for now it's allowed.
+
+    // Iterate through all but the final band (the multipler) and extract the values
+    let value = 0;
+    for (let i = 0; i < (colors.length - 1); ++i) {

```

Comment:
This is where Kyle tells me that for loops are awful and I should use MapReduce or something similar here.